build.gradle

```diff
@@ -79,29 +80,13 @@ pluginBundle {
     }
 }
 
-task releaseNeeded {
-    ext.needed = true
-    doLast {
-        def pullRequest = System.getenv('TRAVIS_PULL_REQUEST')?.matches("\\d+")
-        def onMaster = "master" == System.getenv('TRAVIS_BRANCH')
-        ext.needed = !pullRequest && onMaster
-        if (needed) {
-            logger.lifecycle("{} - Release to Gradle Plugin Portal is needed.", path)
-        } else {
-            logger.lifecycle("{} - Current branch is not 'master' or we're building pull request.\n" +
-                    "  Release to Gradle Plugin Portal is not needed.", path)
-        }
-    }
-}
-
-//TODO ww add -Preleasing.dryRun=false to commandLine when releaseNeeded takes into account [ci skipRelease] in commit msg
 task travisRelease {
     dependsOn releaseNeeded
-    onlyIf { releaseNeeded.needed }
+    onlyIf { !releaseNeeded.releaseNotNeeded }
     doLast {
         logger.lifecycle("{} - Publishing to Gradle Plugin Portal...", path)
         exec {
-            commandLine "./gradlew", "publishPlugins", "bumpVersionAndPush",
+            commandLine "./gradlew", "publishPlugins", "bumpVersionAndPush", "-Preleasing.dryRun=false",
```

Comment:
Excellent!

Now let's try to figure out how to:
1. Hide all that in gradle-plugin-releasing (and also wrap release-needed into that plugin)
2. Avoid forking off Gradle build from inside Gradle build - this makes the build output really hard to grok.